File: python/2/19.py
from collections import defaultdict
import pandas as pd

filename = "../../data/hightemp.txt"
df = pd.read_csv(filename, header=None, delimiter='\t', names=('pref', 'city', 'temp', 'date'))
prefs = defaultdict(int)

for pref in df['pref']:
    prefs[pref] += 1

for pref, count in sorted(prefs.items(), key=lambda x:(-x[1], x[0])):
    print(pref, count)

File: ruby/2/19.rb
require "csv"

filename = "../../data/hightemp.txt"
table = CSV.table(filename, col_sep: "\t", headers: %w(pref city temp day))
prefs = table[:pref].group_by(&:itself).map { |k, v| { "#{k}": v.count } }

prefs.sort_by{ |pref| pref.values }.reverse.each do |hash|
  puts "#{hash.keys.first} #{hash.values.first}"
end
